File: src/talker.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include <sstream>

using namespace ros;

int main(int argc, char **argv) {

    // Initializes node with a name and connects it to master.
    init(argc, argv, "talker");

    // Creates an instance of a node handler object
    NodeHandle n;

    // Initializes the node as a publisher that handles messages of type
    // string, names the topic "chatter", and maximizes the queue to 1000
    Publisher chatter_publisher = n.advertise<std_msgs::String>("chatter", 1000);

    // Sets the execution rate of the processes
    Rate loop_rate(0.5);

    int count = 0;

    // For as long as the node is not shutdown, publishes hello world
    while (ok()) {
        // Sets message data
        std_msgs::String msg;
        std::stringstream ss;
        ss << "Hello World " << count;
        msg.data = ss.str();

        ROS_INFO("[Talker] I published %s\n", msg.data.c_str());

        // Publishes a message to the topic associated with the publisher
        chatter_publisher.publish(msg);

        // Processes callbacks and sleeps for remaining time in cycle
        spinOnce();
        loop_rate.sleep();
        
        count++;
    }

    return 0;
}

File: src/scripts/talker.py
import rospy
from std_msgs.msg import String

def talker():

    """ Initializes a publisher object with topic name chatter and with 
    message type String. Queue size 10 is the amount of messages that can be
    sent asynchronously on different threads """
    pub = rospy.Publisher('chatter', String, queue_size=10)

    """ Initializes a node to connect to the master. Names it 'talker' """
    rospy.init_node('talker', anonymous=True)

    rate = rospy.Rate(1) # Sets how frequently system should sleep to 1 hz

    count = 0

    while not rospy.is_shutdown():
        hello_str = "hello world %s" % count
        rospy.loginfo(hello_str)

        # Publishes message in function param to the topic 'chatter'
        pub.publish(hello_str)

        rate.sleep()

        count += 1

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass